mask-size property refresh (#39418)
mdn/content: files/en-us/web/css/mask-size/index.md

{{CSSRef}}

> The **`mask-size`** [CSS](/en-US/docs/Web/CSS) property specifies the sizes of the mask images. The size of the mask image can be fully or partially constrained in order to preserve its {{glossary("aspect ratio", "intrinsic ratio")}}.

Reviewer: ```suggestion
The **`mask-size`** [CSS](/en-US/docs/Web/CSS) property specifies the sizes of specified mask images. Mask image sizes can be fully or partially constrained to preserve their {{glossary("aspect ratio", "intrinsic aspect ratios")}}.
```

## Syntax

```css

> /* Keywords syntax */

Reviewer: ```suggestion
/* Keyword syntax */
```

This example demonstrates basic usage, while also demonstrating percentage values for `mask-size`.

#### HTML

We include two {{htmlelement("div")}} elements:

```html
<div class="width"></div>
<div class="height"></div>
```

#### CSS

The `<div>` elements are defined to be twice as tall as they are wide, with a gradient background and mask:

```css
div {
  width: 200px;
  height: 400px;
  background: blue linear-gradient(red, blue);
  mask-image: url(https://mdn.github.io/shared-assets/images/examples/mdn.svg);
}
```

The width of one `<div>` element's mask is set to `50%`, with the height defaulting to `auto`. The mask's height for the second `<div>` element is set to `50%` with the width set to `auto`:

```css
.width {
  mask-size: 50%;
}

.height {
  mask-size: auto 50%;
}
```

In the `width` case, the mask is rendered `100px` wide (`50%` of the `200px`-wide element). The height defaults to `auto`, maintaining the mask's aspect ratio.
In the `height` case, the mask is rendered `200px` tall (`50%` of the `400px`-high container). The width is explicitly set to `auto`, maintaining the mask's aspect ratio.

```css hidden
body {
  display: flex;
  gap: 20px;
}
```

#### Results

> {{EmbedLiveSample("Setting mask size as a percentage", "", "430px")}}

Reviewer: The mask image is cropped at the bottom of the first container, and at the right-hand side of the second container, which looks a bit jarring at first. You might want to add a sentence to explain why.
